plugins/modules/digital_ocean_kubernetes.py

```diff
@@ -0,0 +1,438 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2020, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: digital_ocean_kubernetes
+short_description: Create and delete a DigitalOcean Kubernetes cluster
+description:
+  - Create and delete a Kubernetes cluster in DigitalOcean (and optionally wait for it to be running).
+version_added: 1.2.0
+author: Mark Mercado (@mamercad)
+options:
+  oauth_token:
+    description:
+      - DigitalOcean OAuth token; can be specified in C(DO_API_KEY), C(DO_API_TOKEN), or C(DO_OAUTH_TOKEN) environment variables
+    type: str
+    aliases: ['API_TOKEN']
+    required: yes
+  state:
+    description:
+      - The usual, C(present) to create, C(absent) to destroy
+    type: str
+    choices: ['present', 'absent']
+    default: present
+  name:
+    description:
+      - A human-readable name for a Kubernetes cluster.
+    type: str
+    required: yes
+  region:
+    description:
+      - The slug identifier for the region where the Kubernetes cluster will be created.
+    type: str
+    aliases: ['region_id']
+    default: nyc1
+  version:
+    description:
+      - The slug identifier for the version of Kubernetes used for the cluster. See the /v2/kubernetes/options endpoint for available versions.
+    type: str
+    required: no
+    default: 1.18.8-do.1
+  auto_upgrade:
+    description:
+      - A boolean value indicating whether the cluster will be automatically upgraded to new patch releases during its maintenance window.
+    type: bool
+    required: no
```

Comment:
```suggestion
```
Another recommendation is to don't specify default values. Feel free to ignore this one if you prefer to always have them there. But if you want them to be there, better use:
```suggestion
    required: false
```